test/clj/puppetlabs/trapperkeeper/services/webserver/jetty9_default_config_test.clj

```diff
@@ -71,19 +75,110 @@ react accordingly."
+    ;; See: https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-server/src/main/java/org/eclipse/jetty/server/AbstractConnector.java#L146
+    (is (= 30000 (.getIdleTimeout connector))
+        "Unexpected default for 'idle-timeout-milliseconds'")
+    (is (= acceptor-thread-count (.getAcceptors connector))
+        "Unexpected default for 'acceptor-threads' and 'ssl-acceptor-threads'")
+    (is (= selector-thread-count
+           (.getSelectorCount (.getSelectorManager connector)))
+        "Unexpected default for 'selector-threads' and 'ssl-selector-threads'")))
+
+(defn get-max-threads-for-server
+  [server]
+  (.getMaxThreads (.getThreadPool server)))
+
+(defn get-server-thread-pool-queue
+  [server]
+  (let [thread-pool      (.getThreadPool server)
+        ;; Using reflection here because the .getQueue method is protected and I
+        ;; didn't see any other way to pull the queue back from the thread pool.
+        get-queue-method (-> thread-pool
+                             (.getClass)
+                             (.getDeclaredMethod "getQueue" nil))
+        _                (.setAccessible get-queue-method true)]
+    (.invoke get-queue-method thread-pool nil)))
 
 (deftest default-server-settings-test
   (let [server (Server.)]
-    (is (= 30000 (.getStopTimeout server)))))
-
+    ;; See: https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-util/src/main/java/org/eclipse/jetty/util/component/AbstractLifeCycle.java#L48
+    (is (= 30000 (.getStopTimeout server))
+        "Unexpected default for 'shutdown-timeout-seconds'")
+    ;; See: https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-util/src/main/java/org/eclipse/jetty/util/thread/QueuedThreadPool.java#L67
+    (is (= 200 (get-max-threads-for-server server))
+        "Unexpected default for 'max-threads'")
+    ;; See: https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-util/src/main/java/org/eclipse/jetty/util/BlockingArrayQueue.java#L117
+    (is (= (Integer/MAX_VALUE) (.getMaxCapacity
+                                 (get-server-thread-pool-queue server)))
+        "Unexpected default for 'queue-max-size'")))
 
+(def threads-per-connector
+  "The total number of threads needed per attached connector."
+  (+ acceptor-thread-count selector-thread-count))
 
+(defn calculate-minimum-required-threads
+  "Calculate the minimum number threads that a single Jetty Server instance
+  needs.  See: https://github.com/eclipse/jetty.project/blob/jetty-9.2.10.v20150310/jetty-server/src/main/java/org/eclipse/jetty/server/Server.java#L334-L350"
+  [connectors]
+  (+ 1 (* connectors threads-per-connector)))
 
+(deftest default-min-threads-settings-test
+  ;; This test just exists to validate the advice we give for the bare
+  ;; minimum number of threads that one should account for when setting the
+  ;; 'max-threads' setting for a server instance.
```

Comment:
Yeah, just the fact that Jetty calculates its min-threads based on the number of selector and acceptor threads that it needs to allocate.